### ISSUE TITLE
when training KITTI2015, area without disparity ground-truth can't be trained well

### ISSUE THREAD

**Issue by youmi-zym** [JiaRenChang/PSMNet#71]
when I'm training on KITTI2015, the area without disparity ground-truth always can't be trained well, but other area is trained well.For example, blow is the original RGB picture and the disparity map predicted by my algorithm. However, the area of sky get a really bad result.
learning-rate policy is set as yours. here are the epoch=200, the data augment is set as yours.
Please help, thanks!
![screenshot from 2018-08-01 09-59-48](https://user-images.githubusercontent.com/17737537/43496813-dd294464-9571-11e8-8bd2-08038e6c15b8.png)

**Comment by givasile** [comment]
@youmi-zym, In my opinion, this is a normal overfitting problem caused by the lack of such training examples. In general KITTI is a small dataset for training a deep nn (only 200 unique examples) but even more specifically it doesn't contain ground truth for all pixels representing points that are far away from the retina (such as sky points). For this reason, the nn doesn't learn to compare such patches.

**Comment by youmi-zym** [comment]
Is there any good solutions for this issue? such as special loss functions or training trick?

**Comment by youmi-zym** [comment]
@givasile thanks a lot !

**Comment by SamChuah** [comment]
Hi @youmi-zym, did you find a way to solve the overfitting problem? I am facingthe same issue. :(

**Comment by youmi-zym** [comment]
Hi, @SamChuah 
I don't solve it because I found it relates to the robustness of the designed model. So, I'm very sorry to tell you that there is seldom no experience can share with you. 
But, I recommend you refer to https://github.com/DeepMotionAIResearch/DenseMatchingBenchmark/blob/master/configs/PSMNet/ResultOfPSMNet.md#kitti-2015, all training details of PSMNet in KITTI2015 (e.g., loss log, tensor-board log) can be downloaded from here, and you can compare it with your training process.

**Comment by SamChuah** [comment]
Does it mean it does not relate to the hyperparameters like learning rates and training epochs?

**Comment by youmi-zym** [comment]
I'm afraid yes. I had tried with various learning rates and training epochs, but all lead to overfitting. You know, 200 image pairs in KITTI are too small to avoid overfitting. However, for a robust model, the problem I described above happens hardly.

**Comment by SamChuah** [comment]
thank you so much for your help. I still have one doubt. Say if I have a model that can generate relatively good results but with the problem you mentioned above happens in my results, does that mean my model is completely wrong? or it is simply not robust but I can still compare the results with other methods? 

Hope my question makes sense.

**Comment by youmi-zym** [comment]
Sorry, I have no idea about your method and model, so I can hardly provide any advice for you. As for my model, I think the robustness is the core.

**Comment by SamChuah** [comment]
@youmi-zym thank you so much for the advice. You are awesome :)
